### PR TITLE
Prevent unicode on contextual tokens. Fixes #376

### LINKED ISSUE #376: Contextual keywords cannot contain unicode escape sequences
Followup to #307; we're still missing some cases. None of the following are programs, and we accept all of them.

```js
({ g\u0065t x(){} });
```

```js
export {a \u0061s b} from "";
```

```js
export {} fr\u006fm "";
```

```js
for (a o\u0066 b);
```

```js
class a {
  st\u0061tic m(){}
}
```

Gotta be sure to get this right with `async`/`await` too.

### PR DESCRIPTION
**Pull request by Protryon**
Fixes issue #376.

### CODE REVIEW

**Comment by bakkot** [comment]
LGTM, but needs tests.

(Also, you'll need to get lint and tests passing; you can run `npm run lint` and `npm run test` locally.)

**Review by michaelficarra — COMMENTED** [review]
I would simplify this and just mark all tokens where `hasEscape` is `true` as `escaped`. Then we can get rid of `TokenType.ESCAPED_KEYWORD`. Thoughts?

**Comment by bakkot** [comment]
> Then we can get rid of `TokenType.ESCAPED_KEYWORD`.

That would require adding an additional check (for `!token.escaped`) every time we consume any of the many keyword tokens. I don't think it's worth it.

**Comment by michaelficarra** [comment]
Yeah you're probably right. We at least shouldn't mark `ESCAPED_KEYWORD`s with `escaped: false`, though.

**Review by michaelficarra — APPROVED** [review]
LGTM. I believe `escaped` and `hasEscape` are the same thing now, so we can probably just get rid of one of them.

**Comment by bakkot** [comment]
This should at least have the cases in #376 as tests.

**Review by bakkot — APPROVED** [review]

File: src/parser.js [hunk 1]
```diff
@@ -2081,7 +2081,7 @@ export class GenericParser extends Tokenizer {
     if (!isGenerator && token.type === TokenType.IDENTIFIER) {
       if (token.value.length === 3) {
         // Property Assignment: Getter and Setter.
-        if (token.value === 'get' && this.lookaheadPropertyName()) {
+        if (!token.escaped && token.value === 'get' && this.lookaheadPropertyName()) {
```
Comment (michaelficarra) [inline src/parser.js]: Check the things that are more likely to fail earlier. In these cases, `token.value === 'get'` is more likely to be `false` than `!token.escaped`.

File: src/tokenizer.js [hunk 2]
```diff
@@ -804,10 +804,12 @@ export default class Tokenizer {
     let hasEscape = this.index - start !== id.length;
 
     let type = this.getKeyword(id);
-    if (hasEscape && type !== TokenType.IDENTIFIER) {
-      type = TokenType.ESCAPED_KEYWORD;
+    if (hasEscape) {
+      if (type !== TokenType.IDENTIFIER) {
```
Comment (michaelficarra) [inline src/tokenizer.js]: You can `&&` this condition with the previous one to combine these nested `if`s.

**Review by michaelficarra — APPROVED** [review]

**Pull request merged**